Provide the invalid value in the number parsing FormatException message

heathbm opened the pull request:
Implementation of api-approved issue: #65834

heathbm committed:
Provide the invalid value in the number parsing FormatException message

dnfadmin commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=67415) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=67415) before we can accept your contribution.<br/><br/>:x: heathbm [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=67415)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=67415) it.</sub>

stephentoub commented on src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs:
```diff
@@ -2081,15 +2081,15 @@ internal enum ParsingStatus
         internal static void ThrowOverflowException(TypeCode type) => throw GetException(ParsingStatus.Overflow, type);
 
-        private static Exception GetException(ParsingStatus status, TypeCode type)
+        private static Exception GetException(ParsingStatus status, TypeCode type, ReadOnlySpan<char> value = default)
```
Why `value = default`?  That suggests there are some call sites that aren't supplying a value, in which case they're going to end up with a strange `The input string '' was not in a correct format.` message.

heathbm replied:
This has now been addressed in my latest commit

heathbm committed:
Remove default value in the number parsing FormatException message

Clean-up: Number.Parsing.ThrowOverflowException() to call Number.Parsing.GetOverflowException() directly to avoid having to provide a null value for the FormatException message in Number.Parsing.GetException()

heathbm committed:
Merge branch 'main' into more-informational-error-message-when-parsing-number

jeffhandley committed:
Merge branch 'main' into more-informational-error-message-when-parsing-number

jeffhandley reviewed (approved):
Thanks for this contribution, @heathbm!
